scipy/stats/stats.py

```diff
@@ -5581,13 +5586,27 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate'):
         population variance [2]_.
 
         .. versionadded:: 0.11.0
+
     nan_policy : {'propagate', 'raise', 'omit'}, optional
         Defines how to handle when input contains nan.
         The following options are available (default is 'propagate'):
 
           * 'propagate': returns nan
           * 'raise': throws an error
           * 'omit': performs the calculations ignoring nan values
+    permutations : int or None, optional
+        The number of permutations that will be used to calculate p-values
+        using a permutation test. If None, use the t-distribution to calculate
+        p-values; if negative, use all possible permutations of the data.
```

Comment:
1) No. 
2) Nothing protecting against that currently.
3) Yes, each occurs once and only once. I used `itertools`.permutations.

So how about I remove the negative option and if `permutations > n!` I go to the code that runs each once and only once? That makes a lot of sense to me.

Reply:
> So how about I remove the negative option and implement the following:
> 
>     * if `permutations` >= n! I set permutations to n! That is, `permutations = min(permutations, factorial(n)` (or similar)
> 
>     * Later, if `permutations` equals n!, I use the code that runs each permutation exactly once (`indices = np.array(list(permutations(range(m)))).T`)
> 
> 
> This
> 
>     * retains the option to do the exact test
> 
>     * protects against the number of permutations being larger than the maximum number of unique permutations
> 
>     * resolves the issue with permutations being negative below.
> 
> 
> Sound good?

I see, for some reason last night I thought was using `numpy.permutation`. 
This approach sounds reasonable. I think it's worthwhile to be able to distinguish 
the cases when the caller wants a random sample of permutations-w/duplicate arrangements possible-versus when the caller wants an 'exact' test, each permutation occurs once and only once. I know understand we have the exact form, do we also have the random sample form supported as well?

Reply:
Hmm, the more I think about it now the more I prefer your negative approach. That approach permits supporting both the exact test and the resampled/bootstrap test and ***allows*** the bootstrapped test to be used if `# samples > n!`. Sorry for the back and forth on this one. 

I don't see why someone would want to do that but I'm sure their will be reasons/scenarios. 

Maybe leave that part as is and add a check in the statistic calculation beneath to put in the value `n` if `permutations <0`?

Reply:
> So how about I remove the negative option and implement the following:
> 
>     * if `permutations` >= n! I set permutations to n! That is, `permutations = min(permutations, factorial(n)` (or similar)
> 
>     * Later, if `permutations` equals n!, I use the code that runs each permutation exactly once (`indices = np.array(list(permutations(range(m)))).T`)
> 
> 
> This
> 
>     * retains the option to do the exact test
> 
>     * protects against the number of permutations being larger than the maximum number of unique permutations
> 
>     * resolves the issue with permutations being negative below.
> 
> 
> Sound good?

Yes, sounds good to me.

Reply:
OK. `float_factorial` is already used so I went with that. I don't think it needs to be accurate to more than 15 decimal digits because the user has bigger problems if they make `permutations` that large.

Reply:
>OK. `float_factorial` is already used so I went with that. I don't think it needs to be accurate to more than 15 decimal digits because the user has bigger problems if they make permutations that large.

Sounds good, resolving.